Memory assertion when doing backup with 4.7.0

Reading a XML document exceeding 16 MB in size via remote connection triggers the following error:
```
java.lang.AssertionError: memory values bigger than 16MB not supported
	at org.exist.util.io.MemoryContentsImpl.ensureCapacity(MemoryContentsImpl.java:114)
	at org.exist.util.io.MemoryContentsImpl.write(MemoryContentsImpl.java:194)
	at org.exist.util.io.MemoryContentsImpl.writeAtEnd(MemoryContentsImpl.java:216)
	at org.exist.util.io.OverflowToDiskStream.write(OverflowToDiskStream.java:106)
```
The reason for the problem is a typo within the definition of the default in memory buffer of remote resources before buffering them to the local file system of the client. It declares a 64 MB buffer size instead of 4 MB

More details also available on the [Exist-open mailing list entry](https://github.com/eXist-db/exist/issues/2743)

--- a/exist-core/src/main/java/org/exist/xmldb/AbstractRemoteResource.java
+++ b/exist-core/src/main/java/org/exist/xmldb/AbstractRemoteResource.java
@@ -50,7 +50,7 @@ import static org.exist.util.io.InputStreamUtil.copy;
 
 public abstract class AbstractRemoteResource extends AbstractRemote
         implements EXistResource, ExtendedResource, Resource {
-    private static final int DEFAULT_IN_MEMORY_SIZE = 0x400_0000; // 4 MB
+    private static final int DEFAULT_IN_MEMORY_SIZE = 4 * 1024 * 1024; // 4 MB
 
     protected final Leasable<XmlRpcClient>.Lease xmlRpcClientLease;
     protected final XmldbURI path;

--- a/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
+++ b/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
@@ -253,6 +253,7 @@ public class RemoteCollectionTest extends RemoteDBTest {
             Resource res = getCollection().createResource(name, type);
             if(type.equals("XMLResource")) {
                 res.setContent(XML_CONTENT);
+            } else {
                 res.setContent(BINARY_CONTENT);
             }
             getCollection().storeResource(res);

--- a/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
+++ b/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
@@ -9,7 +9,10 @@ import org.exist.xquery.util.URIUtils;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.xml.sax.ContentHandler;
 import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.AttributesImpl;
 import org.xmldb.api.DatabaseManager;
 import org.xmldb.api.base.Collection;
 import org.xmldb.api.base.Resource;
@@ -22,12 +25,14 @@ import org.xmlunit.builder.Input;
 import org.xmlunit.diff.Diff;
 
 import javax.xml.transform.Source;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.StandardCopyOption;
 import java.util.ArrayList;
+import java.util.Arrays;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.junit.Assert.*;
@@ -199,7 +204,7 @@ public class RemoteCollectionTest extends RemoteDBTest {
         assertEquals(0, xmlNames.size());
         assertEquals(0, binaryNames.size());
 	}
-	
+
 	/**
 	 * Trying to access a collection where the parent collection does
 	 * not exist caused NullPointerException on DatabaseManager.getCollection() method.
@@ -218,13 +223,36 @@ public class RemoteCollectionTest extends RemoteDBTest {
         c = DatabaseManager.getCollection(getUri() + colName, "admin", "");
         assertNull(c);
 	}
-	
+
+	@Test  /* issue 2743 */
+    public void getLoadRemoteResourceContentBiggerThan16MB() throws XMLDBException, SAXException {
+        Collection collection = getCollection();
+        final RemoteXMLResource resource = (RemoteXMLResource)collection.createResource("testresource", "XMLResource");
+        prepareContent(resource);
+        collection.storeResource(resource);
+        // load stored content
+        resource.getContentIntoAStream(new ByteArrayOutputStream());
+    }
+
+    private void prepareContent(RemoteXMLResource resource) throws XMLDBException, SAXException {
+        final char[] buffer = new char[1024];
+        Arrays.fill(buffer, (char) 'x');
+        ContentHandler content = resource.setContentAsSAX();
+        content.startDocument();
+        content.startElement("", "root", "root", new AttributesImpl());
+        for (int i = 0, n = 16 * 1024; i < n; i++) {
+            // writing 16 mb to resource
+            content.characters(buffer, 0, 1024);
+        }
+        content.endElement("", "root", "root");
+        content.endDocument();
+    }
+
     private void createResources(ArrayList<String> names, String type) throws XMLDBException {
         for (String name : names) {
             Resource res = getCollection().createResource(name, type);
             if(type.equals("XMLResource")) {
                 res.setContent(XML_CONTENT);
-            } else {
                 res.setContent(BINARY_CONTENT);
             }
             getCollection().storeResource(res);

--- a/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
+++ b/exist-core/src/test/java/org/exist/xmldb/RemoteCollectionTest.java
@@ -231,19 +231,20 @@ public class RemoteCollectionTest extends RemoteDBTest {
         prepareContent(resource);
         collection.storeResource(resource);
         // load stored content
-        resource.getContentIntoAStream(new ByteArrayOutputStream());
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        resource.getContentIntoAStream(outputStream);
+        // compare size
+        assertEquals(16777229, outputStream.size());
     }
 
     private void prepareContent(RemoteXMLResource resource) throws XMLDBException, SAXException {
-        final char[] buffer = new char[1024];
+        final char[] buffer = new char[16 * 1024 * 1024];
         Arrays.fill(buffer, (char) 'x');
         ContentHandler content = resource.setContentAsSAX();
         content.startDocument();
         content.startElement("", "root", "root", new AttributesImpl());
-        for (int i = 0, n = 16 * 1024; i < n; i++) {
-            // writing 16 mb to resource
-            content.characters(buffer, 0, 1024);
-        }
+        // writing 16 mb to resource
+        content.characters(buffer, 0, buffer.length);
         content.endElement("", "root", "root");
         content.endDocument();
     }

